src/sqlfluff/core/parser/lexer.py

```diff
@@ -295,7 +300,7 @@ def __init__(
         )
 
     def lex(self, raw: str) -> Tuple[Tuple[RawSegment, ...], List[SQLLexError]]:
```

Comment:
```suggestion
    def lex(self, raw: Union[str, TemplatedFile]) -> Tuple[Tuple[RawSegment, ...], List[SQLLexError]]:
```
from your docstring below